common/lib/xmodule/xmodule/seq_module.py

```diff
@@ -177,90 +178,153 @@ def handle_ajax(self, dispatch, data):  # TODO: bounds checking
 
         raise NotFoundError('Unexpected dispatch type')
 
+    @classmethod
+    def is_content_visible(cls, due, hide_after_due):
+        """
+        Returns whether the runtime user can view the content
+        of this sequential.
+        """
+        return (
+            not due or
+            not hide_after_due or
+            datetime.now(UTC()) < due
+        )
+
     def student_view(self, context):
+        context = context or {}
+        self._capture_basic_metrics()
+        banner_text = None
+        special_html_view = self._special_exam_student_view() or self._hidden_content_student_view()
+        if special_html_view:
+            masquerading_as_specific_student = context.get('specific_masquerade', False)
+            banner_text, special_html = special_html_view
+            if special_html and not masquerading_as_specific_student:
+                return Fragment(special_html)
+        return self._student_view(context, banner_text)
+
+    def _special_exam_student_view(self):
+        """
+        Checks whether this sequential is a special exam.  If so, returns
+        a banner_text or the fragment to display depending on whether
+        staff is masquerading.
+        """
+        if self.is_time_limited:
+            special_exam_html = self._time_limited_student_view()
+            if special_exam_html:
+                banner_text = _("This exam is hidden from the learner.")
+                return banner_text, special_exam_html
+
+    def _hidden_content_student_view(self):
+        """
+        Checks whether the content of this sequential is hidden from users.
+        If so, returns a banner_text or the fragment to display depending
+        on whether staff is masquerading.
+        """
+        if not self._can_view_content():
+            subsection_format = self.format or _("subsection")  # pylint: disable=no-member
+
+            # Translators: subsection_format refers to the assignment
+            # type of the subsection, such as Homework, Lab, Exam, etc.
+            banner_text = _(
+                "Because the due date has passed, "
+                "this {subsection_format} is hidden from the learner."
+            ).format(subsection_format=subsection_format)  # pylint: disable=no-member
+
+            hidden_content_html = self.system.render_template(
+                'hidden_content.html',
+                {'subsection_format': subsection_format}  # pylint: disable=no-member
+            )
+
+            return banner_text, hidden_content_html
+
+    def _can_view_content(self):
```

Comment:
Good point.  I renamed `_can_view_content` to `_can_user_view_content` and renamed `is_content_visible` to `verify_current_content_visibility`.  I also updated the docstrings to make the distinction clear and correct.  Let me know what you think.